src/tests.rs

```diff
@@ -37,6 +62,166 @@ fn can_deposit() {
+            );
+            assert_noop!(
+                Bank::withdraw(RuntimeOrigin::signed(ALICE), ALICE, 500),
+                Error::<Runtime>::IncorrectRole
+            );
+            assert!(Bank::check_total_issuance());
+        });
+}
+
+#[test]
+fn can_transfer() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000), (BOB, 500)])
+        .build()
+        .execute_with(|| {
+            assert_eq!(
+                Accounts::<Runtime>::get(&ALICE),
+                AccountData {
+                    free: 1_000,
+                    reserved: 0
+                }
+            );
+            assert_eq!(
+                Accounts::<Runtime>::get(&BOB),
+                AccountData {
+                    free: 500,
+                    reserved: 0
+                }
+            );
+            System::reset_events();
+            assert_ok!(Bank::transfer(RuntimeOrigin::signed(ALICE), BOB, 100));
+            // Check that the event was emitted
+            assert_eq!(
+                System::events()[0].event,
+                RuntimeEvent::Bank(Event::<Runtime>::Transfer {
+                    from: ALICE,
+                    to: BOB,
+                    amount: 100
+                })
+            );
+
+            assert_eq!(
+                Accounts::<Runtime>::get(&BOB),
+                AccountData {
+                    free: 600,
+                    reserved: 0
+                }
+            );
+            assert_eq!(
+                Accounts::<Runtime>::get(&ALICE),
+                AccountData {
+                    free: 900,
+                    reserved: 0
+                }
+            );
+            assert_noop!(
+                Bank::transfer(RuntimeOrigin::signed(ALICE), BOB, 1_000),
+                Error::<Runtime>::InsufficientBalance
+            );
+            assert_eq!(Bank::accounts(3), AccountData::default());
+            assert_ok!(Bank::register_role(&3, Role::Manager));
+            assert_noop!(
+                Bank::transfer(RuntimeOrigin::signed(3), BOB, 100),
+                Error::<Runtime>::IncorrectRole
+            );
+            assert!(Bank::check_total_issuance());
+        });
+}
+
+#[test]
+fn can_reaped() {
+    default_test_ext().execute_with(|| {
+        assert_eq!(Bank::accounts(&ALICE), AccountData::default());
+        assert_eq!(Bank::accounts(&BOB), AccountData::default());
+        assert_eq!(Bank::accounts(&3), AccountData::default());
+        assert_ok!(Bank::register_role(&ALICE, Role::Customer));
+        assert_ok!(Bank::register_role(&BOB, Role::Customer));
+        assert_ok!(Bank::register_role(&3, Role::Manager));
+        
+        assert_noop!(Bank::deposit(RuntimeOrigin::signed(3), BOB, 4), Error::<Runtime>::AmountTooSmall);
+        assert_noop!(Bank::withdraw(RuntimeOrigin::signed(3), BOB, 4), Error::<Runtime>::AmountTooSmall);
+        assert_noop!(Bank::transfer(RuntimeOrigin::signed(ALICE), BOB, 4), Error::<Runtime>::AmountTooSmall);
```

Comment:
This top part of the tests are testing amounts smaller than MinimumAmount cannot be used, this should be in a separate test.